src/config/index.ts

```diff
@@ -305,26 +305,40 @@ export class Config extends (EventEmitter as new () => ConfigEventEmitter) {
 	}
 
 	private async generateRequiredFields() {
-		return this.getMany([
+		const {
+			uuid,
+			deviceApiKey,
+			apiSecret,
+			unmanaged,
+			pollIntervalOffset,
+			appUpdatePollInterval,
+		} = await this.getMany([
 			'uuid',
 			'deviceApiKey',
 			'apiSecret',
 			'unmanaged',
-		]).then(({ uuid, deviceApiKey, apiSecret, unmanaged }) => {
-			// These fields need to be set regardless
-			if (uuid == null || apiSecret == null) {
-				uuid = uuid || this.newUniqueKey();
-				apiSecret = apiSecret || this.newUniqueKey();
-			}
-			return this.set({ uuid, apiSecret }).then(() => {
-				if (unmanaged) {
-					return;
-				}
-				if (!deviceApiKey) {
-					return this.set({ deviceApiKey: this.newUniqueKey() });
-				}
-			});
-		});
+			'pollIntervalOffset',
+			'appUpdatePollInterval',
+		]);
+
+		// These fields need to be set regardless
+		if (uuid == null || apiSecret == null) {
+			const newUuid = uuid || this.newUniqueKey();
+			const newApiSecret = apiSecret || this.newUniqueKey();
+			await this.set({ uuid: newUuid, apiSecret: newApiSecret });
+		}
+
+		if (unmanaged) {
+			return;
+		}
+
+		if (!deviceApiKey) {
+			await this.set({ deviceApiKey: this.newUniqueKey() });
+		}
+
+		if (!pollIntervalOffset) {
+			await this.set({ pollIntervalOffset: _.random(appUpdatePollInterval) });
```

Comment:
Am I reading that the offset is going from a random value up to +/-50% of the poll interval calculated per poll, to a random value up to +100% calculated once? So this means a device set at 15m poll interval could in reality use a fixed 30m poll interval, or for a 24h poll interval it could be 48h?